training/benchmarks/glm/pytorch/dataloaders/dataloader.py

```diff
@@ -168,9 +168,10 @@ def build_data_loader(dataset,
     """Data loader. Note that batch-size is the local (per GPU) batch-size."""
     if worker_init_fn is None:
         worker_init_fn = WorkerInitializer.default()
-    world_size = dist.get_world_size()
+    rank = dist_pytorch.get_rank()
+    world_size = dist_pytorch.get_world_size()
     sampler = torch.utils.data.distributed.DistributedSampler(
-        dataset, num_replicas=world_size, shuffle=shuffle)
+        dataset, rank=rank, num_replicas=world_size, shuffle=shuffle)
```

Comment:
Passing in `rank` to `DistributedSampler` , ensuring that `DistributedSampler` will not invoke `torch.distributed.get_rank()` and can still function properly in single device training scenarios where torch distributed group is not initialized.